From e3471b2370601f9251a2d1705712cfb5e3d2316c Mon Sep 17 00:00:00 2001
From: M Safoorah Banu <safoorah@codeaurora.org>
Date: Tue, 16 Oct 2018 10:32:46 +0530
Subject: [PATCH 2/5] BT: API to update quiet mode status in Adapter service
 [1/3]

 - In BLE on use case, quiet mode status is not updated to
   Adapter service which results in not initiating auto
   connection of hfp & a2dp though quiet mode is disabled
   for next turn On.

 - Added new API to update quiet mode status in Adapter
   service.

 Conflicts:
	src/com/android/bluetooth/btservice/AdapterService.java

CRs-Fixed: 2337538
Change-Id: Ifb848e3cc124dd937db09e6d3fbc0996668390ce
Signed-off-by: Volodymyr Zhdanov <wight554@gmail.com>
---
 .../bluetooth/btservice/AdapterService.java      | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index 6c5a7e6f34..c0ed6cafa5 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -2168,6 +2168,16 @@ public class AdapterService extends Service {
             service.mAdapterStateMachine.sendMessage(AdapterState.USER_TURN_ON);
         }
 
+        @Override
+        public void updateQuietModeStatus(boolean quietMode) {
+            AdapterService service = getService();
+            if (service == null) {
+                return;
+            }
+            service.updateQuietModeStatus(quietMode);
+        }
+
+
         @Override
         public void onBrEdrDown() {
             AdapterService service = getService();
@@ -2776,6 +2786,12 @@ public class AdapterService extends Service {
         return mAdapterProperties.getTotalNumOfTrackableAdvertisements();
     }
 
+    void updateQuietModeStatus(boolean quietMode) {
+        debugLog("updateQuietModeStatus()-updateQuietModeStatus called with quiet mode status:"
+                   + quietMode);
+        mQuietmode = quietMode;
+    }
+
     private static int convertScanModeToHal(int mode) {
         switch (mode) {
             case BluetoothAdapter.SCAN_MODE_NONE:
-- 
2.28.0

